packages/sui-indexer/src/handler.ts

```diff
@@ -1,45 +1,77 @@
 import { IndexerStorage } from "./storage";
-import type { MintEventNode, UtxoRecord } from "./models";
+import type {
+	MintEventRaw,
+	ProposeUtxoEventRaw,
+	RedeemRequestEventRaw,
+	SuiEventNode,
+} from "./models";
 import { logger } from "@gonative-cc/lib/logger";
 import type { SuiNet } from "@gonative-cc/lib/nsui";
 
-export async function handleMintEvents(
-	events: MintEventNode[],
+export async function handleEvents(
+	events: SuiEventNode[],
 	storage: IndexerStorage,
 	nbtcPkg: string,
 	suiNetwork: SuiNet,
 ) {
-	const utxosToInsert: UtxoRecord[] = [];
+	for (const event of events) {
+		const json = event.json;
 
-	for (const eventNode of events) {
-		const event = eventNode.json;
+		if (event.type.includes("::MintEvent")) {
+			await handleMint(json as MintEventRaw, storage, nbtcPkg, suiNetwork);
+		} else if (event.type.includes("::RedeemRequestEvent")) {
+			await handleRedeemRequest(json as RedeemRequestEventRaw, storage, nbtcPkg, suiNetwork);
+		} else if (event.type.includes("::ProposeUtxoEvent")) {
+			await handleProposeUtxo(json as ProposeUtxoEventRaw, storage);
+		}
+	}
+}
 
-		const txIdBuffer = Buffer.from(event.btc_tx_id);
-		// NOTE: bitcoin library we use in the other worker uses tx.getId() which returns the reversed order, its just for consistency
-		// TODO: check if we actually need that
-		const txId = txIdBuffer.reverse().toString("hex");
+async function handleMint(e: MintEventRaw, storage: IndexerStorage, pkg: string, net: SuiNet) {
```

Comment:
why don't we have a class? Now every method needs to take basic dependencies as the arguments, rather than grouping it in a class.